calliope/backend/backends.py

```diff
@@ -534,11 +534,18 @@ def _constraint_setter(
             imask: xr.DataArray, expr: tuple[xr.DataArray, str, xr.DataArray]
         ) -> xr.DataArray:
             lhs, op, rhs = expr
+
+            incomplete_constraints = (rhs.isnull() | lhs.isnull()) & imask
+            if incomplete_constraints.any():
+                raise BackendError(
```

Comment:
For consistency with the parsing errors being raised (see #436), it would be good to prefix this error with f"(constraints, {name}): " so that a user need not enter a debugger to find out the name.